### ISSUE TITLE
Uncaught Exception: Cannot create BrowserWindow before app is ready

### ISSUE THREAD

**Issue by rachelmcr** [Automattic/simplenote-electron#361]
#### Steps to reproduce
1. I opened the app from the dock (I think for the first time after restarting my laptop)
2. The app loaded but an error popped up above it
#### What I expected

The app to open.
#### What happened instead

The app opened with this error popup:

```
A JavaScript error occurred in the main process

Uncaught Exception:
Error: Cannot create BrowserWindow before app is ready
    at Error (native)
    at EventEmitter.activateWindow (/Users/rachel/Documents/Simplenote.app/Contents/Resources/app/main.js:29:16)
    at emitTwo (events.js:92:20)
    at EventEmitter.emit (events.js:172:7)
```

After I closed the popup message the app continued to work without any apparent issues.
#### OS version

Mac OS X 10.11.6 (15G31), Simplenote version 1.0.1

**Comment by dmsnell** [comment]
Thanks @rachelmcr. This should be an easily fixable bug.

Has this happened to you more than once? Is there a pattern? If it's a rare fluke we will probably address it later as time permits but if it's happening a bunch and we don't realize it then we can try and take care of it more immediately.

**Comment by rachelmcr** [comment]
Yes, it happens every time I first open the app after restarting my laptop. It isn't a huge deal (I can dismiss the alert and move on) but it is a consistent, recurring thing.

**Comment by roundhill** [comment]
Going to close this for now, haven't seen other reports and we have the native macOS app up to par w/ electron.

**Comment by nefory** [comment]
I know this is late but apparently this error has to do with the way you set your callback to 'ready event.

**Comment by dmsnell** [comment]
@nefory thanks for chiming in! if that's the case as you present it then our lack of types strikes again!

did you notice where this might be going on? I tried to find it but when I looked I was thinking that the error was likely triggered in [desktop/app.js](https://github.com/Automattic/simplenote-electron/blame/a70ddc1e548ae15e0627d91aa0e697504b759964/desktop/app.js#L170) but looking back over the past three years of the call to `app.on( 'ready', … )` I couldn't find the proposed mistake.

where did you find it?
